src/Layer.js

```diff
@@ -378,9 +378,9 @@ export default class Layer {
     }
 
     _addToMGLMap(map, beforeLayerID) {
-        if (map.isStyleLoaded()) {
+        try {
             this._onMapLoaded(map, beforeLayerID);
-        } else {
+        } catch (error) {
```

Comment:
since multiple errors could happen here I'd look for the specific mapbox error and act accordingly

Reply:
I agree with that, but I am not sure how to check it since `error` is a string and the message may vary also. This is how the error is triggered

```js 
    _checkLoaded() {
        if (!this._loaded) {
            throw new Error('Style is not done loading');
        }
    }
```

Reply:
```js
catch(err) {
// We can control the situation when the error is "style not done loading"
  if(err.message === 'Style is not done loading') {
    this._onMapLoaded(map, beforeLayerID);
    return;
  }

// Otherwise a "real" error happened, forward it
throw error;
}

Reply:
done :)